src/runtime/recipe/recipe.ts

```diff
@@ -582,5 +582,15 @@ export class Recipe {
     });
   }
 
+  getParticlesByImplFile(particleSet) {
+    return this.particles.filter(particle => particle.spec && particleSet.has(particle.spec.implFile));
+  }
+
+  findSlotByID(id) {
+    return this.slots.find(s => s.id === id);
+  }
 
+  getMatchingConnections() {
```

Comment:
The name is a bit confusing; I realise you've based it on where it was originally used, but as an API it's not clear what "matching" means, and the filter() below doesn't seem to be matching anything.

It looks like the variable name "matchingConnections" ultimately refers to the output of the map() in match-free-handles-to-connections.ts. That logic should remain there, so perhaps this method should be called `getUnconnectedConnections`.

Reply:
I removed "getMatchingConnections" altogether and instead use "getDisconnectedConnections" instead (an existing method that essentially does the same thing).

Reply:
:+1: